bindings/azure/signalr/signalr.go

```diff
@@ -0,0 +1,195 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package signalr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/gbrlsnchs/jwt/v3"
+	"github.com/pkg/errors"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	errorPrefix         = "azure signalr error:"
+	logPrefix           = "azure signalr:"
+	connectionStringKey = "connectionString"
+	hubKey              = "hub"
+	groupKey            = "group"
+	userKey             = "user"
+)
+
+// NewSignalR creates a new pub/sub based on Azure SignalR
+func NewSignalR() *SignalR {
+	return &SignalR{
+		tokens:     make(map[string]signalrCachedToken),
+		httpClient: &http.Client{Timeout: 30 * time.Second},
+	}
+}
+
+type signalrCachedToken struct {
+	token      string
+	expiration time.Time
+}
+
+// SignalR is an output binding for Azure SignalR
+type SignalR struct {
+	endpoint   string
+	accessKey  string
+	version    string
+	hub        string
+	tokens     map[string]signalrCachedToken
+	httpClient *http.Client
+}
+
+// Init is responsible for initializing the SignalR output based on the metadata
+func (s *SignalR) Init(metadata bindings.Metadata) error {
+	connectionString, ok := metadata.Properties[connectionStringKey]
+	if !ok || connectionString == "" {
```

Comment:
for my edification, `connectionString == ""` or `len(connectionString) == 0`? I see both patterns in this file.

Reply:
According to what I [read](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go) the compiler will do make both [equal](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go/54804858#54804858), so we should use the one that is more readable.

Will change to == as I find more readable